test/e2e/source_binding_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 The Knative Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or im
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+// +build !serving
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"gotest.tools/assert"
+	"knative.dev/client/pkg/util"
+)
+
+func TestSourceBinding(t *testing.T) {
+	t.Parallel()
+	test := NewE2eTest(t)
+	test.Setup(t)
+	defer test.Teardown(t)
+
+	test.serviceCreate(t, "testsvc0")
+
+	t.Run("create source binding", func(t *testing.T) {
+		test.sourceBindingCreate(t, "my-binding0", "Deployment:apps/v1:myapp", "svc:testsvc0")
+	})
+
+	t.Run("delete source binding", func(t *testing.T) {
+		test.sourceBindingDelete(t, "my-binding0")
+	})
+
+	t.Run("update source binding", func(t *testing.T) {
+		test.sourceBindingCreate(t, "my-binding1", "Deployment:apps/v1:myapp", "svc:testsvc0")
+		test.serviceCreate(t, "testsvc1")
+		test.sourceBindingUpdate(t, "my-binding1", "Deployment:apps/v1:myapp", "svc:testsvc1")
+		out, err := test.kn.RunWithOpts([]string{"source", "binding", "describe", "my-binding1"}, runOpts{NoNamespace: false})
+		assert.NilError(t, err)
+		assert.Check(t, strings.Contains(out, "testsvc1"))
```

Comment:
I wonder whether we have the option to check whether there has been actually a server-side `SinkBinding` CO has been created. The test looks good, but tbh the value is not much more than within the unit test which checks the actual output on the console for the command, too.

IMO any E2E test should verify, ideally with other channels than with `kn` itself, whether the operation has been successfully performed.

So my preference for verifying a test would be:

* As the CO be created/updated with the content expected ?
* Or: If this is not possible or to difficult, then use e.g `kn binding describe` to check for it. E.g. one coudl do a `kn binding describe my-binding1 -o json`, parse the output an verify its content. This would at least give some enhanced certainty that the object has been created/update properly.

@MIBc do you think you could switch to `kn binding describe -o json` here ?

Reply:
kn source binding describe do not support `-o`.
How about `kubectl get sinkbindings.sources.knative.dev my-binding -ojsonpath="{.spec.sink.ref.name}"`
I find the `getResourceFieldsWithJSONPath` can do this.